meson.build

```diff
@@ -123,6 +123,14 @@ if cc.compiles(set_prio_code, name : 'test for setpriority support')
   conf_data.set('HAVE_SETPRIORITY', 1)
 endif
 
+# Except for macOS, Clang requires explicit linking with the atomic lib
+if cxx.get_id() == 'clang' and host_machine.system() != 'darwin'
+  atomic_dep = cc.find_library('atomic', required: true)
+  if atomic_dep.found()
+    add_project_link_arguments('-latomic', language: 'cpp')
+  endif
+endif
+
```

Comment:
This is weird… On my system, my small test programs compile and link fine without `-latomic` using both GCC and Clang, but code in this PR does not compile… I suspect there's a different cause for this link failure and we should not be adding this meson snippet in here.

Perhaps some field in native file (`.github/meson/native-clang.ini`) is missing?

Reply:
Yes, it's the CI systems that need this. 
If you remove this, you will get the latomic missing symbols link error and CI will fail.

On my home system, I'm like you and also don't need the explicit latomic. 

You mention > but code in this PR does not compile…

Oh, strange? Can you paste the error?

Reply:
OK, I still don't know what exactly is the issue but found some reports that latomic needs to be linked in some GCC versions as well.

Let's remove this code snippet completely. Replace it with:

In place where other dependencies are listed:
```meson
atomic_dep = cxx.find_library('atomic')
```
And then add `atomic_dep` to the list of dependencies in `src/misc/meson.build`.

Reply:
Pushed a commit replacing prior w/ `atomic_dep = cxx.find_library('atomic')`, 

macOS clang failing:

https://github.com/dosbox-staging/dosbox-staging/pull/922/checks?check_run_id=2033017394

``` text
The Meson build system
Version: 0.57.1
Source dir: /Users/runner/work/dosbox-staging/dosbox-staging
Build dir: /Users/runner/work/dosbox-staging/dosbox-staging/build
Build type: native build
Project name: dosbox-staging
Project version: 0.77.0
C compiler for the host machine: ccache cc (clang 12.0.0 "Apple clang version 12.0.0 (clang-1200.0.32.29)")
C linker for the host machine: cc ld64 609.8
C++ compiler for the host machine: ccache c++ (clang 12.0.0 "Apple clang version 12.0.0 (clang-1200.0.32.29)")
C++ linker for the host machine: c++ ld64 609.8
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -Wextra-semi: YES 
Checking for function "clock_gettime" : YES 
Checking for function "mprotect" : YES 
Checking for function "realpath" : YES 
Checking whether type "struct dirent" has member "d_type" : YES 
Has header "pwd.h" : YES 
Has header "strings.h" : YES 
Has header "netinet/in.h" : YES 
Has header "sys/socket.h" : YES 
Checking if "test for setpriority support" compiles: YES 
Checking if "test for __builtin_expect support" compiles: YES 

meson.build:151:0: ERROR: C++ shared or static library 'atomic' not found

A full log can be found at /Users/runner/work/dosbox-staging/dosbox-staging/build/meson-logs/meson-log.txt
Error: Process completed with exit code 1.
```

Ubuntu 20.04 failing:

https://github.com/dosbox-staging/dosbox-staging/pull/922/checks?check_run_id=2033017621

``` text
 [193/206] Linking target tests/soft_limiter.
FAILED: tests/soft_limiter 
clang++  -o tests/soft_limiter 'tests/59830eb@@soft_limiter@exe/soft_limiter.cpp.o' 'tests/59830eb@@soft_limiter@exe/stubs.cpp.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group src/misc/libmisc.a /usr/bin/../lib/gcc/x86_64-linux-gnu/10/../../../x86_64-linux-gnu/libgtest_main.a -lpthread /usr/bin/../lib/gcc/x86_64-linux-gnu/10/../../../x86_64-linux-gnu/libgtest.a /usr/lib/x86_64-linux-gnu/libSDL2.so -Wl,--end-group '-Wl,-rpath,$ORIGIN/../src/misc' -Wl,-rpath-link,/home/runner/work/dosbox-staging/dosbox-staging/build/src/misc
/usr/bin/ld: src/misc/libmisc.a(soft_limiter.cpp.o): in function `std::atomic<AudioFrame>::load(std::memory_order) const':
/usr/bin/../lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/atomic:258: undefined reference to `__atomic_load'
/usr/bin/ld: src/misc/libmisc.a(soft_limiter.cpp.o): in function `std::atomic<AudioFrame>::store(AudioFrame, std::memory_order)':
/usr/bin/../lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/atomic:247: undefined reference to `__atomic_store'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[194/206] Compiling C++ object 'tests/59830eb@@string_utils@exe/string_utils.cpp.o'.
clang: warning: -lpthread: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: -Wl,--start-group: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: -Wl,--end-group: 'linker' input unused [-Wunused-command-line-argument]
[195/206] Compiling C++ object 'tests/59830eb@@setup@exe/setup.cpp.o'.
clang: warning: -lpthread: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: -Wl,--start-group: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: -Wl,--end-group: 'linker' input unused [-Wunused-command-line-argument]
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.
```

Reply:
@dreamer - it might make sense for you to take-over this aspect the PR and find the more prefered Meson solution for including libatomic?

Reply:
How about `cxx.find_library('atomic', required : false)`?

Reply:
``` python
optional_dep  = dependency('', required : false)
atomic_dep    = cxx.find_library('atomic', required : false)
threads_dep   = dependency('threads')
```

Getting closer!

macOS CI is happy now: https://github.com/dosbox-staging/dosbox-staging/pull/922/checks?check_run_id=2034413149

![2021-03-04_12-21](https://user-images.githubusercontent.com/1557255/110025462-41027580-7ce4-11eb-9766-68dd17a83306.png)

Ubuntu 20.04 Clang finds `atomic` at setup:

![2021-03-04_12-24](https://user-images.githubusercontent.com/1557255/110025693-93dc2d00-7ce4-11eb-9b15-c5f6c59656ae.png)

But bombs out during linking:

![2021-03-04_12-20](https://user-images.githubusercontent.com/1557255/110025514-537caf00-7ce4-11eb-9561-836ee36ad01e.png)

Presumably because it's not actually passing `-latomic` (atleast I can't see it). 

https://github.com/dosbox-staging/dosbox-staging/pull/922/checks?check_run_id=2034413290

Reply:
Trying again with `dependencies : [**atomic_deb**, threads_dep, sdl2_dep]`

Reply:
We're good now! Resolving.